injectproxy/routes.go

```diff
@@ -301,22 +311,41 @@ func NewRoutes(upstream *url.URL, label string, extractLabeler ExtractLabeler, o
 		)
 	}
 
-	errs.Add(
-		// Reject multi label values with assertSingleLabelValue() because the
-		// semantics of the Silences API don't support multi-label matchers.
-		mux.Handle("/api/v2/silences", r.el.ExtractLabel(
-			enforceMethods(
-				assertSingleLabelValue(r.silences),
+	if opt.regexMatch {
```

Comment:
(suggestion) to avoid duplicating the mux.Handle() calls, we might do something like this:

```go
errs.Add(
    mux.Handle("/api/v2/silences", r.errorIfRegexpMatch(
        r.el.ExtractLabel(...

[...]

 func (r *routes) errorIfRegexpMatch(next http.HandlerFunc) http.HandlerFunc {
    return func(w http.ResponseWriter, req *http.Request) {
        if r.regexMatch {
            prometheusAPIError(w, "support for regex match not implemented", http.StatusNotImplemented)
            return
        }

        return next(w, req)
    }
}

```

Reply:
Done @simonpasquier. Back to you ⚾ !